pkg/cloud/azure/actuators/machine/actuator.go

```diff
@@ -100,16 +101,22 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 	if err != nil {
 		return a.handleMachineError(machine, apierrors.CreateMachine("failed to create machine %q scope: %v", machine.Name, err), createEventAction)
 	}
-	defer scope.Close()
 
 	err = a.reconcilerBuilder(scope).Create(context.Background())
 	if err != nil {
+		if err := scope.Store(); err != nil {
```

Comment:
Why do try and persist in face of an error?

Reply:
Because the failed status conditions are set

Reply:
> Because the failed status conditions are set

That's not very obvious. Can we capture we we do this. TIA